Compiler warning in Bytes.h: casts away qualifiers

### Check duplicate issues.

- [x] Checked for duplicates

### Description

/root/root/include/Bytes.h: In function ‘void tobuf(char*&, ULong_t)’:
/root/root/include/Bytes.h:109:15: warning: cast from type ‘const ULong_t*’ {aka ‘const long unsigned int*’} to type ‘char*’ casts away qualifiers [-Wcast-qual]
  109 |    char *sw = (char *)intermediary;
      |               ^~~~~~~~~~~~~~~~~~~~
/root/root/include/Bytes.h: In function ‘void tobuf(char*&, Long_t)’:
/root/root/include/Bytes.h:148:15: warning: cast from type ‘const Long_t*’ {aka ‘const long int*’} to type ‘char*’ casts away qualifiers [-Wcast-qual]
  148 |    char *sw = (char *)intermediary;
      |               ^~~~~~~~~~~~~~~~~~~~


In both cases, "intermediary" is defined as "const", which is indeed cast away by the C-style cast. As far as I can see, "sw" isn't written to, so that probably should be "const" as well? Otherwise, a const_cast could be an option.

### Reproducer

If you try to compile anything that (indirectly) includes include/Bytes.h with -Wcast-qual enabled.

### ROOT version

6.32.04

### Installation method

Pre-built binary

### Operating system

AlmaLinux 9.4

### Additional context

_No response_

--- a/core/base/inc/Bytes.h
+++ b/core/base/inc/Bytes.h
@@ -71,7 +71,7 @@ inline void tobuf(char *&buf, UShort_t x)
 # else
    // To work around a stupid optimization bug in MSVC++ 6.0
    const UShort_t *intermediary = &x;
-   char *sw = (char *) intermediary;
+   const char *sw = (const char *) intermediary;
    buf[0] = sw[1];
    buf[1] = sw[0];
 # endif
@@ -89,7 +89,7 @@ inline void tobuf(char *&buf, UInt_t x)
 # else
    // To work around a stupid optimization bug in MSVC++ 6.0
    const UInt_t *intermediary = &x;
-   char *sw = (char *)intermediary;
+   const char *sw = (const char *)intermediary;
    buf[0] = sw[3];
    buf[1] = sw[2];
    buf[2] = sw[1];
@@ -106,7 +106,7 @@ inline void tobuf(char *&buf, ULong_t x)
 #ifdef R__BYTESWAP
    // To work around a stupid optimization bug in MSVC++ 6.0
    const ULong_t *intermediary = &x;
-   char *sw = (char *)intermediary;
+   const char *sw = (const char *)intermediary;
    if (sizeof(ULong_t) == 8) {
       buf[0] = sw[7];
       buf[1] = sw[6];
@@ -145,7 +145,7 @@ inline void tobuf(char *&buf, Long_t x)
 #ifdef R__BYTESWAP
    // To work around a stupid optimization bug in MSVC++ 6.0
    const Long_t *intermediary = &x;
-   char *sw = (char *)intermediary;
+   const char *sw = (const char *)intermediary;
    if (sizeof(Long_t) == 8) {
       buf[0] = sw[7];
       buf[1] = sw[6];
@@ -201,7 +201,7 @@ inline void tobuf(char *&buf, ULong64_t x)
 # else
    // To work around a stupid optimization bug in MSVC++ 6.0
    const ULong64_t *intermediary = &x;
-   char *sw = (char *)intermediary;
+   const char *sw = (const char *)intermediary;
    buf[0] = sw[7];
    buf[1] = sw[6];
    buf[2] = sw[5];

